src/Selenium2Library/keywords/_browsermanagement.py

```diff
@@ -315,7 +323,16 @@ def select_window(self, locator=None):
         | Title Should Be | Popup Title |
         | Select Window |  | | # Chooses the main window again |
         """
-        self._window_manager.select(self._current_browser(), locator)
+        try:
+            return self._current_browser().get_current_window_handle()
```

Comment:
So, I might be missing something. If they provide a locator, couldn't this possible return the wrong window?